eZ/Publish/Core/settings/repository.yml

```diff
@@ -149,4 +150,11 @@ services:
 
     ezpublish.signalslot.signal_dispatcher:
         class: %ezpublish.signalslot.signal_dispatcher.class%
+        factory: ["@ezpublish.signalslot.signal_dispatcher.factory", buildSignalDispatcher]
 
+    ezpublish.signalslot.signal_dispatcher.factory:
+        class: %ezpublish.signalslot.signal_dispatcher.factory.class%
+        arguments:
+            - "%ezpublish.signalslot.signal_dispatcher.class%"
+            - "$repository$"
```

Comment:
This should not be here, since we don't know about `$repository$` or `%ezpublish.repositories%` at this level.

I propose you implement a `Core` version of `SignalDispatcherFactory`, which would receive search engine alias directly. Then you can extend that in the `EzPublishCoreBundle` and add knowledge about `$repository$` and `%ezpublish.repositories%`.
Maybe implement internal method `getSearchEngineAlias()`, in `Core` is can simply return injected alias, and in the bundle it can resolve the same from injected repository alias and repository settings (you will also have to override this service definition in the bundle).

Then you can also lose emulation of the full app by adding `ezpublish.repositories` param to the container builder.